plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiComponentRegistrar.kt

```diff
@@ -20,6 +21,9 @@ class JvmAbiComponentRegistrar : ComponentRegistrar {
     override fun registerProjectComponents(project: MockProject, configuration: CompilerConfiguration) {
         val outputPath = configuration.getNotNull(JvmAbiConfigurationKeys.OUTPUT_PATH)
         if (configuration.get(JvmAbiConfigurationKeys.LEGACY_ABI_GEN, false)) {
+            require(!configuration.getBoolean(CommonConfigurationKeys.USE_FIR)) {
```

Comment:
Throwing exceptions from compiler is a bad practice. Please replace this `require` with reporting an error to message collector